### PR TITLE
New: Azure DNS support

### PR DESCRIPTION
**Pull request by RobinDadswell**
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/docker-swag/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description:

Adds support for Azure DNS as a DNS plugin
## Benefits of this PR and context:

Anyone that uses Azure DNS as their DNS provider will be able to utilise DNS-01 challenges
## How Has This Been Tested?

Tested container built by Jenkins (lspipepr/swag:1.23.0-PR-217) - gets certificate as expected using Azure DNS for wildcard certificate, proxy services work with this. 

## Source / References:

Rox pointed me in the right places on Discord in messages starting here https://discord.com/channels/354974912613449730/805754854982025246/946532138088542208

### CODE REVIEW

File: root/defaults/dns-conf/azure.ini [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#dns_azure_zone1 = example.com:/subscriptions/c135abce-d87d-48df-936c-15596c6968a5/resourceGroups/dns1
+
+# Azure Managed Identity
+#dns_azure_msi_client_id = 912ce44a-0156-4669-ae22-c16a17d34ca5
+#dns_azure_zone1 = example.com:/subscriptions/c135abce-d87d-48df-936c-15596c6968a5/resourceGroups/dns1
+
+# System Assigned Azure Managed Identity
+#dns_azure_msi_system_assigned = true
+#dns_azure_environment = "AzurePublicCloud"
+#dns_azure_zone1 = example.com:/subscriptions/c135abce-d87d-48df-936c-15596c6968a5/resourceGroups/dns1
```
Comment (RobinDadswell) [inline root/defaults/dns-conf/azure.ini:26]: all example config has come from the docs site for certbot-dns-azure

**Comment by RobinDadswell** [comment]
This is waiting for feedback/merge and isn't with me so shoild be excluded from the Bot

**Comment by RobinDadswell** [comment]
@Roxedus  can you get this excluded from stale bot? It's on someone at LSIO to look at and provide feedback :)

**Comment by aptalca** [comment]
missing other dockerfiles and incorrectly inserted into init file

**Pull request merged**